osu.Game/Overlays/Mods/ModColumn.cs

```diff
@@ -0,0 +1,409 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using Humanizer;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Utils;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+#nullable enable
+
+namespace osu.Game.Overlays.Mods
+{
+    public class ModColumn : CompositeDrawable
+    {
+        private Func<Mod, bool>? filter;
+
+        public Func<Mod, bool>? Filter
+        {
+            get => filter;
+            set
+            {
+                filter = value;
+                updateFilter();
+            }
+        }
+
+        private readonly ModType modType;
+        private readonly Key[]? toggleKeys;
+
+        private readonly Bindable<Dictionary<ModType, IReadOnlyList<Mod>>> availableMods = new Bindable<Dictionary<ModType, IReadOnlyList<Mod>>>();
+
+        private readonly TextFlowContainer headerText;
+        private readonly Box headerBackground;
+        private readonly Container contentContainer;
+        private readonly Box contentBackground;
+        private readonly FillFlowContainer<ModPanel> panelFlow;
+        private readonly ToggleAllCheckbox? toggleAllCheckbox;
+
+        private Colour4 accentColour;
+
+        private const float header_height = 42;
+
+        public ModColumn(ModType modType, bool allowBulkSelection, Key[]? toggleKeys = null)
+        {
+            this.modType = modType;
+            this.toggleKeys = toggleKeys;
+
+            Width = 320;
+            RelativeSizeAxes = Axes.Y;
+            Shear = new Vector2(ModPanel.SHEAR_X, 0);
+            CornerRadius = ModPanel.CORNER_RADIUS;
+            Masking = true;
+
+            Container controlContainer;
+            InternalChildren = new Drawable[]
```

Comment:
I'm always a bit hesitant to have larger drawable hierarchies constructed in the `ctor` as it moves the async requirement one level higher. But I can see how this works better for nullability concerns (at the cost of the colour assignments getting uglier, though).

Probably fine until we run into actual performance issues, but at some point in a distant past we did have a general consensus on doing this in BDL where possible.

Reply:
I'm stuck between a rock and a hard place here, because previously [it was deemed unfavourable to have `null!`-marked fields that aren't DI'd](https://github.com/ppy/osu/pull/16264#discussion_r775506750). Moving the construction to BDL allows to eliminate some of the fields (which are only there for colour application's sake) but there are some like `panelFlow` that aren't removable. So I'm not sure where to take this because as far as I can see I have two dead ends and a third bad option of asserting everywhere I guess?

Reply:
I think it’s fine as is, was just pointing out because it’s come up when O review your stuff, but a likely non-issue.

Optimally we could move DI also to ctor in some future and have the best of everything.